hydromt/utils.py

```diff
@@ -65,3 +70,15 @@ def _dict_pprint(d):
     import json
 
     return json.dumps(d, indent=2)
+
+
+def has_no_data(
+    data: Optional[Union[pd.DataFrame, gpd.GeoDataFrame, xr.Dataset, xr.DataArray]],
+) -> bool:
+    """Check whether various data containers are empty."""
```

Comment:
Also in the docstings :) Not sure if this part of exposed API or just internal though, if internal then typehint is enough for other hydromt developpers.